clients/java/src/test/java/io/camunda/zeebe/client/group/AssignUserToGroupTest.java

```diff
@@ -0,0 +1,84 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.group;
+
+import static io.camunda.zeebe.client.impl.http.HttpClientFactory.REST_API_PATH;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.github.tomakehurst.wiremock.verification.LoggedRequest;
+import io.camunda.zeebe.client.api.command.ProblemException;
+import io.camunda.zeebe.client.protocol.rest.ProblemDetail;
+import io.camunda.zeebe.client.util.ClientRestTest;
+import io.camunda.zeebe.client.util.RestGatewayService;
+import org.junit.jupiter.api.Test;
+
+public class AssignUserToGroupTest extends ClientRestTest {
+
+  private static final long USER_KEY = 123L;
+  private static final long GROUP_KEY = 456L;
+
+  @Test
+  void shouldAssignUserToGroup() {
+    // when
+    client.newAssignUserToGroupCommand(USER_KEY, GROUP_KEY).send().join();
+
+    // then
+    final LoggedRequest request = RestGatewayService.getLastRequest();
+    assertThat(request.getUrl())
+        .isEqualTo(REST_API_PATH + "/groups/" + GROUP_KEY + "/users/" + USER_KEY);
+    assertThat(RestGatewayService.getLastRequest().getMethod().getName()).isEqualTo("POST");
+  }
+
+  @Test
+  void shouldRaiseExceptionOnNotFoundGroupOrUser() {
+    // given
+    gatewayService.errorOnRequest(
+        REST_API_PATH + "/groups/" + GROUP_KEY + "/users/" + USER_KEY,
+        () -> new ProblemDetail().title("Not Found").status(404));
+
+    // when / then
+    assertThatThrownBy(() -> client.newAssignUserToGroupCommand(USER_KEY, GROUP_KEY).send().join())
+        .isInstanceOf(ProblemException.class)
+        .hasMessageContaining("Failed with code 404: 'Not Found'");
+  }
+
+  @Test
+  void shouldHandleServerError() {
+    // given
+    gatewayService.errorOnRequest(
+        REST_API_PATH + "/groups/" + GROUP_KEY + "/users/" + USER_KEY,
+        () -> new ProblemDetail().title("Internal Server Error").status(500));
+
+    // when / then
+    assertThatThrownBy(() -> client.newAssignUserToGroupCommand(USER_KEY, GROUP_KEY).send().join())
+        .isInstanceOf(ProblemException.class)
+        .hasMessageContaining("Failed with code 500: 'Internal Server Error'");
+  }
+
+  @Test
+  void shouldRaiseExceptionOnForbiddenRequest() {
+    // given
+    gatewayService.errorOnRequest(
+        REST_API_PATH + "/groups/" + GROUP_KEY + "/users/" + USER_KEY,
+        () -> new ProblemDetail().title("Forbidden").status(403));
+
+    // when / then
+    assertThatThrownBy(() -> client.newAssignUserToGroupCommand(USER_KEY, GROUP_KEY).send().join())
+        .isInstanceOf(ProblemException.class)
+        .hasMessageContaining("Failed with code 403: 'Forbidden'");
+  }
```

Comment:
💭 I wonder if there's much point in testing this as a client unit test. All this is testing is the generic client error handling, since you you tell the gateway exactly what error should be thrown. Imo this has way more value as an integration test. Which I see you've already added 👍

Reply:
@remcowesterhoud do you think I should remove these tests?

Reply:
Yeah, unless you think otherwise I'm fine with them being gone.